x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/optimizer/OptimizerRules.java

```diff
@@ -1509,6 +1513,104 @@ protected Expression regexToEquals(RegexMatch<?> regexMatch, Literal literal) {
         }
     }
 
+    // a IS NULL AND a IS NOT NULL -> FALSE
+    // a IS NULL AND a > 10 -> a IS NULL
+    // can be extended to handle null conditions where available
+    public static class PropagateNullable extends OptimizerExpressionRule {
```

Comment:
Isn't this also changing the way a projection is now returned when it comes to conjunctions with `NULL`able and `IS NULL` expressions? `SELECT LENGTH(NULL) > 10 AND NULL IS NULL` returns now `NULL`, after this PR I think is returning `TRUE`.

Reply:
Good point - the rule is too aggressive and needs adjusting and only return null as there's another one that can pick up and reduce the null accordingly.